tests/test_memory.py

```diff
@@ -0,0 +1,9 @@
+from devito import DenseData
+import pytest
+import numpy as np
+
+
+@pytest.mark.parametrize('shape', [(20, 20), (20, 20, 20), (20, 20, 20, 20)])
+def test_first_touch(shape):
```

Comment:
The `first_touch` function does not manipulate ASTs, unlike the DLE method for cache blocking. `first_touch` is a one-line unit that only deals in `Operator`s and arrays - which is being tested by this unit test (see updated test). I see no reason to complicate the function for this reason alone.